diff -ruN lldb-14.0.0.src.orig/source/Plugins/Language/CPlusPlus/CPlusPlusLanguage.cpp lldb-14.0.0.src/source/Plugins/Language/CPlusPlus/CPlusPlusLanguage.cpp
--- a/lldb-14.0.0.src.orig/source/Plugins/Language/CPlusPlus/CPlusPlusLanguage.cpp	2022-03-14 10:44:55.000000000 +0100
+++ b/lldb-14.0.0.src/source/Plugins/Language/CPlusPlus/CPlusPlusLanguage.cpp	2022-06-08 07:06:32.227607437 +0200
@@ -940,11 +940,6 @@
   stl_deref_flags.SetFrontEndWantsDereference();
 
   cpp_category_sp->GetRegexTypeSyntheticsContainer()->Add(
-      RegularExpression("^std::vector<.+>(( )?&)?$"),
-      SyntheticChildrenSP(new ScriptedSyntheticChildren(
-          stl_synth_flags,
-          "lldb.formatters.cpp.gnu_libstdcpp.StdVectorSynthProvider")));
-  cpp_category_sp->GetRegexTypeSyntheticsContainer()->Add(
       RegularExpression("^std::map<.+> >(( )?&)?$"),
       SyntheticChildrenSP(new ScriptedSyntheticChildren(
           stl_synth_flags,
